### ISSUE TITLE
Eyetracker on pixel 4xl runs slowly

### ISSUE THREAD

**Issue by eissa** [issalab/mkturk#41]
Noticed that eyetracker sample interval on the mobile phone (pixel  4xl) is quite large (>20ms => <45Hz sample rate).

**Comment by eissa** [comment]
We decided to turn off both eyetracker and display timing logging since these are fairly high bandwidth timeseries. We also turned off sample command issuing. The eyetracker reads still remained slow.

The only thing left  was  scene display. We found that commenting out render3D caused the eyetracker reads to speed up. More specifically, commenting out renderer.render within render3D was causing the slowdown. Commenting out update3D did not have a noticeable effect.

**Comment by eissa** [comment]
Solution: threeJSrenderratio was set to 3.5x, but we now prefer to use 2x. So having a smaller render canvas and fixing movieframerate to 30Hz instead of 45Hz on phone cause eyetracker timing to be much faster and comparable to the PC.

**Comment by eissa** [comment]
In order to be robust to rendering many objects and ever more complicated scenes, we decided to move rendering to the desktop GPU. This is done by using the pixel 4xl simply as an external display connected to our homebrew PC workstation with an NVIDIA 2080Ti card. This hardware configuration gives us much faster rendering so that mkturk can keep up with the incoming eye stream. Also, we are using the USB port on the PC for communicating with the arduino which is a more reliable WebUSB than what we've had on the Android OS (Android 11+ seem to not work with webUSB giving horrendous eyetracker sampling speeds; Android 10 does, so we froze OS version on the pixel 4xl).
Bottom line, faster rendering & faster WebUSB are strong reasons to run mkturk from a PC when available even if the subject is ultimately doing the task on a mobile device (SuperXDisplay app let's us use Android devices as external displays with low display & touch interactivity lags).

**Comment by eissa** [comment]
Note, perhaps in the future when OffscreenCanvas works reliably, we can use mobile to do rendering without affecting other processes so strongly. However, given the finicky behavior of both OffscreenCanvas & WebUSB on mobile (but not on desktop testing, I believe), our hands were tied, and it's just much more straightforward to develop & run mkturk on desktop when possible.
